[9.0][MIG] quality_control

LoisRForgeFlow opened the pull request:
Continue the works started in https://github.com/OCA/manufacture/pull/100.

oihane committed:
Modulos de calidad sin migrar

oihane committed:
quality_control * PEP8 * some fixing * translations

pedrobaeza committed:
quality_control * PEP8 * Travis * Size in field (deprecated)

alfredoavanzosc committed:
no msg

pedrobaeza committed:
quality_control: Code enhanced

alfredoavanzosc committed:
no msg

pedrobaeza committed:
quality_control: Some improvement

oihane committed:
Flake8 failing fixed

alfredoavanzosc committed:
no msg

oihane committed:
<quality_control> deleted constraint

alfredoavanzosc committed:
no msg

oihane committed:
quality_control: * Pylint fixing * PEP8 fixing

alfredoavanzosc committed:
no msg

oihane committed:
quality_control: * Made line tree editable on test templates

* Fixing problems

pedrobaeza committed:
quality_control: Total refactorization with new API, README files, and new concepts. * Triggers for product category * ACLs * Tolerances in questions * one2many copyable * Fix some views * Error on same tolerance * Colors on inspection lines * Unify inspection header data assignment in set_test method * key name in set_test * Change icon set

oihane committed:
New module <quality_control_hr> closes #662 * Changes in access rules * Trigger line instead of test and user

pedrobaeza committed:
quality_control * Fix wizard set test

* Expand AUTHORS

* Fix permissions

oihane committed:
Fixed email address

ejans committed:
changed default references of fields.Datetime.now() to fields.Datetime.now

sysadminmatmoz committed:
Slovene translations added

oihane committed:
quality_control: * Changed permissions

* security modified
* some demo info added
* Added "partner_selectable" field, so that inspections can by made by partner
* Little view fixing, field was oe_edit_only instead of label
* Add partners to trigger lines
* Tests added
* Trigger lines with duplicated test cleaned

oihane committed:
quality_control: Propose to OCA

oihane committed:
quality_control: required changes

oihane committed:
quality_control: Fixing coveralls

oihane committed:
quality_control_stock: * added demo data * replaced unlink method by ondelete="cascade" * required changes

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
quality_control_*: Make use of qty field

Making it a regular field, you can put any manual quantity in inspections

oca-transbot committed:
OCA Transbot updated translations from Transifex

oihane committed:
quality_control: deleting qc.inspection does not delete its lines fixed

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
remove en.po that was erroneously created by transbot

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
quality_control: decimal precision configurable

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
quality_control: approving inspections only quality control managers

oca-transbot committed:
OCA Transbot updated translations from Transifex

LoisRForgeFlow committed:
[9.0][MIG] quality_control

Commit:
[9.0][FIX] quality_control: not take into account inactivated tests trigger lines

MiquelRForgeFlow reviewed (approved):
LGTM :+1: 
But I think some of the previous commits could be squashed.

lreficent commented:
Cherry-picked fix done in https://github.com/OCA/manufacture/pull/205/commits

LoisRForgeFlow commented:
Cherry-picked fix done in https://github.com/OCA/manufacture/pull/205

andhit-r commented on quality_control/wizard/qc_test_wizard.py:
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r commented on quality_control/wizard/__init__.py:
```diff
@@ -0,0 +1,5 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r commented on quality_control/models/qc_trigger_line.py:
```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r commented on quality_control/models/qc_trigger.py:
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r commented on quality_control/models/qc_test_category.py:
```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r commented on quality_control/models/qc_test.py:
```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r commented on quality_control/models/qc_inspection.py:
```diff
@@ -0,0 +1,295 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r commented on quality_control/models/product_template.py:
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r commented on quality_control/models/product_product.py:
```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r commented on quality_control/models/product_category.py:
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r commented on quality_control/models/__init__.py:
```diff
@@ -0,0 +1,13 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r commented on quality_control/__init__.py:
```diff
@@ -0,0 +1,7 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
```
Change into short header

andhit-r reviewed (changes requested):
Code review

LoisRForgeFlow committed:
headers

Commit:
[IMP] quality_control: show action_cancel button in inspections even if it is autogenerated

LoisRForgeFlow commented:
Open PRs in v8:
* [x] : https://github.com/OCA/manufacture/pull/193
* [x] : https://github.com/OCA/manufacture/pull/213 (merged and added [here](https://github.com/OCA/manufacture/pull/198/commits/784cce1fed74758f8042f56c1d181dc53baa51cf))

LoisRForgeFlow commented:
@andhit-r Is this ok for you now? could you update your review?

LoisRForgeFlow commented:
@andhit-r ping

LoisRForgeFlow commented:
@OCA/manufacturing-maintainers can we get an update on this?

damendieta commented:
Hi, on creation of the Test Categories there there is an error that tries to compute the name but as It's a new record, the value is still false:

File "/mnt/extra-addons/manufacture/quality_control/models/qc_test_category.py", line 23, in _get_complete_name
    self.complete_name = " / ".join(reversed(names))
TypeError: sequence item 0: expected string, bool found

Replacing:
names = [self.name]
With:
names = [self.name or '']

Solves the issue.

Daniel.

damendieta commented:
Hi, I already fixed it and added some minor changes that I needed to test the module.

I made a PR with the changes:
https://github.com/Eficent/manufacture/pull/2

Also, could you please explain the logic of triggers on the readme, I coudn't find a way to make triggers work.

Thanks a lot.

LoisRForgeFlow commented:
Hi @damendieta 

First of all: thanks for your review!

Regarding the bug you reported, it is already being fixed in v8 and waiting to be ported to this PR after its merge, as noted here: https://github.com/OCA/manufacture/pull/198#issuecomment-312675900. Please review that PR (https://github.com/OCA/manufacture/pull/193) in order to move forward.

About the qc triggers, let me clarify that to make use of them you need either [quality_control_mrp](https://github.com/OCA/manufacture/tree/8.0/quality_control_mrp) or [quality_control_stock](https://github.com/OCA/manufacture/tree/8.0/quality_control_stock) modules. However, they still need to be migrated to v9.

Your help would be appreciated :wink:

damendieta commented:
@lreficent ok, I approved the PR #193, what else do we need for it to get merged?

zamberjo committed:
[FIX] quality_control: Fix TypeError on new test category (#193)

`TestQualityControl` is failing on `QcTestTemplateCatergory._get_complete_name`
when `name` is set as `False`.

LoisRForgeFlow commented:
@damendieta fix from https://github.com/OCA/manufacture/pull/193 added. Now you can review this PR and approve it if everything is ok for you.

eLBati commented:
@lreficent  hi, could you check commits like https://github.com/OCA/manufacture/pull/198/commits/e6d89619eeab5355d9c60887851c055b4a048acb with no message and unrecognised author?
Also please check if some commits can be squashed

LoisRForgeFlow commented:
Hi @eLBati  

It looks like "alfredo" has changed either his github email or name and that's why the author is unrecognized, however I don't know his current github account (If someone could point it to me I'll change the commit author accordingly). Regarding the squashing, I've already squashed consecutive transifex commits, I don't know if I can squash further since I'm not the author of any of the previous versions commits.

eLBati commented:
Maybe @oihane or @pedrobaeza know about his account?

oihane commented:
Alfredo de la Fuente's github user is alfredoavanzosc

LoisRForgeFlow commented:
@oihane Thanks!

@eLBati history has been fixed.

eLBati commented on quality_control/views/product_category_view.xml:
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
```
Please use odoo

LoisRForgeFlow replied:
Done, Thanks!

eLBati reviewed (approved):
Just a remark

SimoRubi reviewed (commented):
Just code review, looks good

LoisRForgeFlow committed:
[9.0][FIX] quality_control: use odoo tag on views

LoisRForgeFlow commented:
@SimoRubi Hi, please use the option to approve when submitting the review:
![selection_206](https://user-images.githubusercontent.com/23449160/33068279-000a4bc8-ceb1-11e7-994e-66900151dc62.png)

LoisRForgeFlow commented:
@damendieta If you are ok with this we can merge it I think

SimoRubi reviewed (approved):
Code review, looks good
